sdk/core/azure-core-amqp/inc/azure/core/amqp/internal/common/runtime_context.hpp

```diff
@@ -31,16 +30,23 @@ namespace Azure { namespace Core { namespace Amqp { namespace Common { namespace
 
   using UniqueRustRuntimeContext = Azure::Core::Amqp::_detail::UniqueHandleHelper<
       Azure::Core::Amqp::_detail::RustRuntimeContext>::type;
-  class RustThreadContext final {
+
+  /**
+   * @brief Represents the an implementation of the rust multithreaded runtime.
+   *
+   * Needed to implement blocking Rust API calls.
+   */
+  class RustRuntimeContext final {
 
     UniqueRustRuntimeContext m_runtimeContext;
 
   public:
-    RustThreadContext()
-        : m_runtimeContext(Azure::Core::Amqp::_detail::RustInterop::runtime_context_new())
+    RustRuntimeContext()
+        : m_runtimeContext{Azure::Core::Amqp::_detail::RustInterop::runtime_context_new()}
     {
     }
-    Azure::Core::Amqp::_detail::RustRuntimeContext* GetRuntimeContext() const noexcept
+
+    Azure::Core::Amqp::_detail::RustRuntimeContext* GetRuntimeContext()
```

Comment:
Why is returning direct pointers okay here?

Reply:
Because we are never transferring ownership of the underlying pointer, so it is safe to return a raw pointer. The only time you see a raw pointer returned is when it is a non-reference case.